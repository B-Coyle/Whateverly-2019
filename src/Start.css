
.startWrapper {
    display: block;
    position: relative;
    background: url("./WhateverlyImgs/StartBackgrnd.jpeg") no-repeat;
    background-size: cover;
    margin: 0;
    /* z-index: 2; */
}

.campStartBtn,
.hikeStartBtn {
    display: block;
    width: 10%;
    height: 400px;
    color: black;
    background-color: white;
}

/* .visibility {
    visibility: hidden;
} */

.startLocationSearch {
   display: block;
}

/* .searchVisibility {
    
} */

.startLocationInput {
    width: 40%;
    height: 100px;
    border: 5px solid red;
    color: black;
}

.startSubmitBtn {
    width: 10%;
    height: 50px;
    border: 5px solid blue;
    color: black;
    background-color: green;
}