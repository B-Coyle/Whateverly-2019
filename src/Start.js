import React, {Component} from 'react';
import App from './App.js';

class Start extends App {
  constructor() {
    super();

  }
  render {
    return (
      
    )
  }
}

export default Start;