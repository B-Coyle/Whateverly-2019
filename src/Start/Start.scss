

.header {
    text-align: center;
    font-size: 3em;
    color: white;
}

.startWrapper {
    margin: 0;
    padding: 0;
    width: 100%;
    grid-column-start: 2
}

.startLocationInput {
    width: 40%;
    height: 60px;
    border: 3px solid #af5700 ;
    color: black;
    margin: 0 30%;
    margin-bottom: 3%;
}

.startLocationSearch {
    display: flex;
    margin-top: 3%;
    width: 100%;
    flex-wrap: wrap;
    justify-content: space-evenly;
}

.buttonHolder {
    display: flex;
    flex-wrap: wrap;
    justify-content: space-evenly;
    width: 40%;
}

.campStartBtn,
.hikeStartBtn {
    width: 40%;
    height: 50px;
    color: rgb(230, 228, 237);
    background-color: rgb(62, 39, 47);
    border: 1px solid white;
}

.campStartBtn:hover {
    color: #e8dddd;
    background-color: #af5700;
}

.hikeStartBtn:hover {
    color: #e8dddd;
    background-color: #af5700;
}

