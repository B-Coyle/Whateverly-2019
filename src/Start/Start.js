import React, {Component} from 'react';
import './Start.css';


export default class Start extends Component {
  constructor(props) {
    super(props);
      this.state= {
        startInputValue: ''
      }
    }

    handleChange = (event) => {
      event.preventDefault();
      this.setState({
        startInputValue: event.target.value
      }) 
    }

    handleClickCampingBtn = (props) => {
      // console.log('hello');
      this.props.userCampInput(this.state.startInputValue);
      this.props.filteredCampground(this.state.startInputValue);
      this.props.campingBoolean();
    }

    handleClickHikingBtn = (props) => {
      console.log('HI');
      this.props.userHikeInput(this.state.startInputValue);
      this.props.filteredHiking(this.state.startInputValue);
      this.props.hikingBoolean();
    }

  
  render () {
    return (
      <section className='startWrapper'>
        <h1 className="header">XCO</h1>
        <section className='startLocationSearch'>
          <input type="text" className="startLocationInput" placeholder="Please Enter a Location..." value= {this.state.startInputValue} onChange={this.handleChange} />
          <div className='buttonHolder'>
          <input type="button" className="campStartBtn"  value='Camping' onClick={this.handleClickCampingBtn}  />
          <input type="button" className="hikeStartBtn" value='Hiking' onClick={this.handleClickHikingBtn} />
        </div>
        </section>
      </section>
    )
  }
}


