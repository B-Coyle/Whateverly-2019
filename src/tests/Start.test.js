import React from 'react';
import ReactDOM from 'react-dom';
import Start from '../Start/Start.js';
import { shallow } from 'enzyme';


const mockCamping = [
    {
    "name": "Glacier Basin Campground",
    "location": "Estes Park",
    "season": "May-Sept",
    "elevation": 8500,
    "camping": ["Tent", "Car", "RV"],
    "activities": ["Hiking", "Fishing", "Horseback Riding", "Wildlife"],
    "reservation-available": true,
    "nearest-city": null
      },
      {
    "name": "Grand Junction KOA",
    "location": "Grand Junction",
    "season": "All year",
    "elevation": 4600,
    "camping": ["Tent", "Car", "RV"],
    "activities": ["ATVing"],
    "reservation-available": true,
    "nearest-city": null
    }
]
const mockInput = 'Grand Junction';

const mockHandleChange = jest.fn();


describe('Start', () => {
let wrapper 

beforeEach(() => {
    wrapper = shallow(
        <Start />
    )
});
it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

it('should have a default state', () => {
    expect(wrapper.state()).toEqual({
        startInputValue: ''
    })
});

// it('should have an event listener for handle change', () => {
//     expect(mockHandleChange).toBeCalled()
// });

it('should have an event listener for handle click', () => {
    const mockClickCampingBtn = jest.fn();
    const wrapper = shallow(<Start onClick = {mockClickCampingBtn}/>)
    wrapper.find('input').hasClassName('campStartBtn').simulate('click');
    expect(mockClickCampingBtn).toHaveBeenCalled()
});

})

//Need to review testing and test out based on functionality with App