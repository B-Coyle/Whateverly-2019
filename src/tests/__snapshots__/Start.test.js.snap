// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Start should mock snapshot 1`] = `undefined`;
